cpp/velox/memory/VeloxMemoryPool.cc

```diff
@@ -338,25 +401,32 @@ class WrappedVeloxMemoryPool : public facebook::velox::memory::MemoryPool {
     }
   }
 
+  const int64_t cap_;
+  MemoryManager& memoryManager_;
+
   // Memory allocated attributed to the memory node.
   facebook::velox::memory::MemoryUsage localMemoryUsage_;
   std::shared_ptr<facebook::velox::memory::MemoryUsageTracker> memoryUsageTracker_;
   mutable folly::SharedMutex subtreeUsageMutex_;
   facebook::velox::memory::MemoryUsage subtreeMemoryUsage_;
-  int64_t cap_;
   std::atomic_bool capped_{false};
 
-  std::shared_ptr<Allocator> allocator_;
+  facebook::velox::memory::MemoryAllocator& allocator_;
+  std::shared_ptr<VeloxMemoryAllocatorVariant> glutenAllocator_;
 };
 
 std::shared_ptr<facebook::velox::memory::MemoryPool> AsWrappedVeloxMemoryPool(MemoryAllocator* allocator) {
-  return std::make_shared<WrappedVeloxMemoryPool<VeloxMemoryAllocatorVariant>>(
-      "wrapped", nullptr, std::make_shared<VeloxMemoryAllocatorVariant>(allocator));
+  static IMemoryManager::Options options;
+  static MemoryManager manager{options};
```

Comment:
Since we don't return a static instance from the function, is it OK to pass the static arguments into each instance?

Reply:
So the design is to make the `MemoryManager` singleton, then pass it to each pool, do I understand it correctly?

Reply:
I use static just because the example here uses memory on the stack.
https://github.com/facebookincubator/velox/blob/main/velox/common/memory/tests/MemoryPoolTest.cpp#L82-L85
And I think static have a longer lifespan.

Reply:
@zhejiangxiaomai no need to address my comment if you designed it as singleton, I saw the `MemoryManager` is just a reference in Velox pool.